eclipse-collections/src/main/java/org/eclipse/collections/impl/map/mutable/UnmodifiableMutableMap.java

```diff
@@ -1148,22 +1148,4 @@ public MutableShortBag collectShort(ShortFunction<? super V> shortFunction)
     {
         return this.getMutableMap().zipWithIndex();
     }
-
```

Comment:
Similar to the comment above. This is the more standard pattern for implementing unmodifiable behavior.

Reply:
Catching up on your other comments, I take the point about it being more difficult to perform the code review when I remove methods, and it requires reasoning about whether that's ok. I have to point out that this is not that much of a standard pattern for implementing unmodifiable behavior. I just tried going into UnmodifiableMutableMap and pressing Cmd+I (to implement, not override) and I see many methods implemented that are otherwise not implemented using this pattern.

* flatCollectWith
* flatCollectWith (target)
* countBy
* countBy(target)
* countByWith
* countByWith(target)
* countByEach
* countByEach(target)
* forEach(Procedure)
* forEach(Consumer)
* getAny
* minOptional()
* maxOptional()
* minOptional(Comparator)
* maxOptional(Comparator)
* minByOptional
* maxByOptional
* summarizeInt
* summarizeFloat
* summarizeLong
* summarizeDouble
* reduceInPlace(Collector)
* reduceInPlace(Supplier, BiConsumer)
* reduce
* groupByAndCollect

Reply:
Looking at this list of methods, most of them were probably added as default methods, which we started doing after the 8.0 release. I saw a few since 9.0 and 10.0 marks in the JavaDoc. It's an interesting question whether their behavior should be overridden in `UnmodifiableMutableMap`. If you pull this method up, then you should probably pull all of the API's up. If we agree this makes sense, I would rather this be done as a separate PR for `UnmodifiableMutableMap`.

All of these older methods are implemented directly in `UnmodifiableMutableMap`, just as `aggregateBy` is today.

```
    public <R> MutableBagMultimap<R, V> groupBy(Function<? super V, ? extends R> function)
    public <R, C extends MutableMultimap<R, V>> C groupBy(Function<? super V, ? extends R> function, C target)
    public <R> MutableBagMultimap<R, V> groupByEach(Function<? super V, ? extends Iterable<R>> function)
    public <R, C extends MutableMultimap<R, V>> C groupByEach(Function<? super V, ? extends Iterable<R>> function, C target)
    public <VV> MutableMap<VV, V> groupByUniqueKey(Function<? super V, ? extends VV> function)
    public <VV, R extends MutableMapIterable<VV, V>> R groupByUniqueKey(Function<? super V, ? extends VV> function, R target)
```

Reply:
Yes, I plan on pulling all those up and more. I started with a few "easy" methods to see how it goes.

Reply:
Ok, I think I can get comfortable with the unmodifiable moves here.